cg-spec/editors_draft.html

```diff
@@ -901,6 +1327,99 @@ <h2>Translate Basic Graph Patterns</h2>
 
     </section>
 
+    <section id="builtin-functions">
+      <h2>Function Definitions</h2>
+
+      <p>SPARQL introduces operators and functions that can be used in an expression of a FILTER clause, a BIND clause, or a SELECT clause (see <a data-cite="SPARQL11-QUERY#OperatorMapping">Section&nbsp;17.3</a> and <a data-cite="SPARQL11-QUERY#SparqlOps">Section&nbsp;17.4</a> in [[SPARQL11-QUERY]]). While these operators and functions are defined to operate on <a>RDF terms</a> and <a>query variables</a>, to use them also for SPARQL-star they have to be defined to operate on <a>RDF-star terms</a> and <a>query variables</a>. To this end, when using these operators and functions in the context of SPARQL-star, their definitions as given in <a data-cite="SPARQL11-QUERY#SparqlOps">Section&nbsp;17.4</a> of [[SPARQL11-QUERY]] are extended by assuming that any mention of <code>RDF term</code> as a data type for operands is understood to be the type of all <a>RDF-star terms</a>.</p>
+
+      <p>In addition to carrying over the operators and functions of SPARQL, SPARQL-star introduces five new functions that are defined as follows (where the data types <code>RDF-star term</code> and <code>RDF-star triple</code> capture all <a>RDF-star terms</a> and all <a>RDF-star triples</a>, respectively).</p>
+
+      <section>
+        <h2>TRIPLE</h2>
+
+        <p>
+          <code>RDF-star triple</code> &nbsp;
+          <code style="color:black;font-weight:bold">TRIPLE</code>
+          (
+            <code>RDF-star term </code>
+            <code style="color:black">term1</code>,
+            <code>RDF-star term </code>
+            <code style="color:black">term2</code>,
+            <code>RDF-star term </code>
+            <code style="color:black">term3</code>
+          )
+        </p>
+
+        <p>If the 3-tuple (<code style="color:black">term1</code>, <code style="color:black">term2</code>, <code style="color:black">term3</code>) is an <a>RDF-star triple</a>, the function returns this triple. If the 3-tuple is not an RDF-star triple, then the function raises an error.</p>
```

Comment:
not a _valid_ RDF-star triple?

Reply:
When I wrote this sentence, I also wanted to use include the word "valid" at first. However, there is no notion validity of RDF-star triples. I think the sentence is okay as it is.  The notion of an RDF-star triple is well defined and, thus, there is no ambiguity in what this sentence means.